app/plugin_install.go

```diff
@@ -16,12 +17,65 @@ import (
 	"github.com/mattermost/mattermost-server/utils"
 )
 
+// managedPluginFileName is the file name of the flag file that marks
+// a local plugin folder as "managed" by the file store.
+const managedPluginFileName = ".filestore"
+
+// fileStorePluginFolder is the folder name in the file store of the plugin bundles installed.
+const fileStorePluginFolder = "plugins"
+
+func (a *App) InstallPluginFromData(data model.PluginEventData) {
+	mlog.Debug("Installing plugin from cluster message", mlog.String("plugin_id", data.Id))
+
+	fileStorePath := a.getBundleStorePath(data.Id)
+	reader, appErr := a.FileReader(fileStorePath)
+	if appErr != nil {
+		mlog.Error("Failed to open plugin bundle from filestore.", mlog.String("path", fileStorePath), mlog.Err(appErr))
+	}
+	defer reader.Close()
+
+	if _, appErr = a.installPluginLocally(reader, true); appErr != nil {
+		mlog.Error("Failed to unpack plugin from filestore", mlog.Err(appErr), mlog.String("path", fileStorePath))
+	}
+}
+
+func (a *App) RemovePluginFromData(data model.PluginEventData) {
+	mlog.Debug("Uninstalling plugin from cluster message", mlog.String("plugin_id", data.Id))
+
+	if err := a.removePluginLocally(data.Id); err != nil {
+		mlog.Error("Failed to remove plugin locally", mlog.Err(err), mlog.String("id", data.Id))
```

Comment:
Depending on which we we flip:

```suggestion
		mlog.Error("Failed to uninstall plugin locally", mlog.Err(err), mlog.String("id", data.Id))
```

Reply:
+1 for `uninstall`

Reply:
Discussed offline, but I think everywhere pretty much uses `remove`, so we'd be making things worse in terms of uniformity by introducing `uninstall` at this level. I've flipped a few comments back to `remove` as such.